### PR TITLE
Feat: Sign Out User

### PR DESCRIPTION
**Pull request by nzikak**
Sign out now signs the user out at the backend and removes the email, session token, and user ID shared prefs on the app.
This is the complete implementation of the Signout workflow and addresses issue #827.

Files Changed/Added:
- sign_out_view.dart
- sign_out_viewmodel.dart
- organization_view.dart

Video:

https://user-images.githubusercontent.com/31522978/134801640-c003e86b-a767-4878-8448-edeecef2cde0.mp4

### CODE REVIEW

File: lib/ui/view/sign_out/sign_out_viewmodel.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/cupertino.dart';
```
Comment (Blazebrain) [inline lib/ui/view/sign_out/sign_out_viewmodel.dart:1]: There should be no material or Cupertino import in the ViewModel

Kindly adjust and fix

**Review by Blazebrain — CHANGES_REQUESTED** [review]
Very good work

Fix requested changes and we good to go

File: lib/ui/view/sign_out/sign_out_viewmodel.dart (hunk 1 above)
Comment (nzikak) [reply lib/ui/view/sign_out/sign_out_viewmodel.dart]: I've done that. @Blazebrain

**Comment by YenumJay** [comment]
Pull latest changes from dev and resolve conflicts.

File: lib/ui/view/organization/organization_view/organization_view.dart [hunk 2]
```diff
@@ -161,10 +161,11 @@ class OrganizationTile extends ViewModelWidget<OrganizationViewModel> {
           style: AppTextStyles.body3Medium,
         ),
       ),
-      trailing: GestureDetector(
-        onTap: () {},
-        child: const Icon(Icons.more_vert),
-      ),
+      trailing: GestureDetector(onTap: () {
+        showModalBottomSheet(context: context,
```
Comment (pauleke65) [inline lib/ui/view/organization/organization_view/organization_view.dart:165]: Please use the bottom sheet service and call it from the model

Comment (pauleke65) [inline lib/ui/view/organization/organization_view/organization_view.dart:165]: Please use the bottom sheet service and call it from the model

File: lib/ui/view/sign_out/sign_out_view.dart [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class BottomSheetContent extends ViewModelWidget<SignOutViewModel> {
```
Comment (pauleke65) [inline lib/ui/view/sign_out/sign_out_view.dart:86]: Use the bottom sheet service

File: lib/ui/view/sign_out/sign_out_view.dart [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+              Icons.logout_sharp,
+              color: AppColors.zuriPrimaryColor,
+            ),
+          ),
+        )
+      ],
+    );
+  }
+
+  Future<void> _showSignOutDialog(context, String orgName, SignOutViewModel model) async {
```
Comment (pauleke65) [inline lib/ui/view/sign_out/sign_out_view.dart:121]: Use the Dialog service

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Great work.
- Just fix requested changes
- run ```flutter analyze``` and make sure there are no issue
and you're good to go

**Comment by pauleke65** [comment]
@nzikak please make requested changes so that your PR can be merged

**Review by pauleke65 — APPROVED** [review]

**Review by jeremiahseun — APPROVED** [review]

**Review by YenumJay — APPROVED** [review]
LGTM

**Pull request merged**